### PR TITLE
dev-cpp/sdbus-c++, gui-libs/gtk-layer-shell, gui-apps/waybar update

### PR DESCRIPTION
**Pull request by JohnMH**
Update dev-cpp/sdbus-c++, dependency of several Hypr* packages.

Update gui-apps/waybar and gui-libs/gtk-layer-shell as dep.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Review by chewi — CHANGES_REQUESTED** [review]
`src_install` doesn't even work for sdbus-c++ because it fails on removing `COPYING`. `src_compile` also fails with `USE=doc`. Did you try this package at all? :confused:

File: dev-cpp/sdbus-c++/sdbus-c++-2.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+EAPI=8
+
+inherit meson cmake
+
+DESCRIPTION="High-level C++ D-Bus library"
+HOMEPAGE="https://github.com/Kistler-Group/sdbus-cpp"
+SRC_URI="https://github.com/Kistler-Group/sdbus-cpp/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="LGPL-2.1+ Nokia-Qt-LGPL-Exception-1.1" # Nothing to do with Qt but exception text is exactly the same.
+SLOT="0/1"
```
Comment (chewi) [inline dev-cpp/sdbus-c++/sdbus-c++-2.0.0.ebuild:12]: Also the SLOT needs to change to `0/2`.

**Comment by chewi** [comment]
> `src_install` doesn't even work for sdbus-c++ because it fails on removing `COPYING`.

Sorry, I do take that part back. That only happens when `USE=test`. That's very weird.

**Comment by JohnMH** [comment]
I didn't test with USE=doc.

As soon as my systems are done running updates I'll take a look.

**Comment by chewi** [comment]
I've been working through the issues. It's just the failing tests I haven't figured out yet. I'm heading to bed, but I'd be grateful if you could figure that part out. Sorry again for the misunderstanding. I'm a little grumpy because I don't actually need this package any more. :upside_down_face: Nothing in the repo depends on it.

**Comment by chewi** [comment]
It seems to be because `DBUS_SESSION_BUS_ADDRESS` isn't making it through to the ebuild environment. I can't work out what's killing it, but `sudo` and such would kill it anyway. We probably need to run the tests via `dbus-run-session` or something.

**Comment by JonasToth** [comment]
https://github.com/gentoo/gentoo/pull/39025
there is a pr for waybar already

**Comment by JohnMH** [comment]
@JonasToth Removed 0.11.0. Removed wlroots dep from live and double checked the other deps. Live wlroots will ned gtk-layer-shell 0.9.0.

**Review by lanodan — APPROVED** [review]

**Comment by chewi** [comment]
I've got the tests passing, but I need sleep now. Will pull this together tomorrow.

**Pull request closed without merging**